### ISSUE TITLE
Open File from Integrated Terminal in Current Instance 

### ISSUE THREAD

**Issue by denisrheault** [microsoft/vscode#118718]
From the Integrated Terminal, one should be able to open a file from the command line in the current instance (when running multiple instances of vs code).

Currently the --reuse-window opens the file in the last active window but this is not the wanted behavior. The wanted behavior is much like when ctrl-clicking on a file in the integrated terminal; it opens in the current instance of code. 

I have multiple instances of VS code running - one for each projects. Using --reuse-window opens the file in an instance of code that is not related to the project.

**Comment by denisrheault** [comment]
No, not covered in existing issues.
Denis

**Comment by yume-chan** [comment]
> ... should be able to open a file from the command line in the current instance
> ... --reuse-window opens the file in the last active window

How do you open a file from Integrated Terminal of an inactive window?

**Comment by denisrheault** [comment]
Why open in the last active window? It should open in the current window.
For me using the switch --reuse-window or no switch has the same effect.
Normal?

Say I open a project in instance A and a second project in instance B.
In the interminal terminal of instance A I do 'code -r fileA' the file will
open in the last active window (project B) even though it is for project A.
Doesn't make sense.
If I ctrl-click (open selected) of fileA it opens in project A.  This is
what I want, but there are things I can do from command line but not with
ctrl-click. Like a bash function that finds the selected file and opens it
once the directory is found.

**Comment by denisrheault** [comment]
Any progress on this? Is it clear what I am asking for?  Thanks in advance.

**Comment by Tyriar** [comment]
Can you elaborate a little on what you mean here? If you open a file from the CLI, it will open it in the most relevant windows AFAICT. See this gif how I open xterm.js/README.md from the vscode window, but it opens in the correct window:

![recording (17)](https://user-images.githubusercontent.com/2193314/113038103-f6d9ac00-914a-11eb-928a-4cfacf59d565.gif)

**Comment by denisrheault** [comment]
Hi Tyriar,

From the CLI, I want the same behavior as when clicking "Open File in Editor (ctrl + click)"; the file opens in the current window (same as the integrated terminal).

I realized while trying to replicate to make a GIF is that the behavior is as expected as long as I don't SSH into another server.

Without SSH'ing, it actually works as expected, as shown in this GIF. In the GIF I show that in the integrated terminal I can hover a filename (xrun.log) and ctrl-click'ing opens the file in the current window.
![vsCode_CLI_ok](https://user-images.githubusercontent.com/36093362/113420700-bfac0a80-9397-11eb-84ef-918e7b54db79.gif)

I repeated the same thing, first showing that 'code ./xrun.log' still opens the file in the current window, next SSH into another server and opens the same file (it's a shared drive therefore file exists on both servers).
Unexpectedly another Window opens xrun.log.  Now let's try hovering on the filename and ctrl-click'ing opens the file in the current window as expected.

![vsCode_CLI_NOTok](https://user-images.githubusercontent.com/36093362/113420755-d9e5e880-9397-11eb-821d-fc71b7e8ec61.gif)

Now that I understand that the problem is with SSH and not being able to open a file in a window (process) that is running on a different machine.
I don't think there's anything that could be done in this case. Let me know what you think.
Thanks!
